tools/reviewer/job/sync/GitHubWriter.java

```diff
@@ -0,0 +1,150 @@
+
+package com.google.startupos.tools.reviewer.job.sync;
+
+import com.google.startupos.tools.reviewer.localserver.service.Protos.Comment;
+import com.google.startupos.tools.reviewer.localserver.service.Protos.Diff;
+import com.google.startupos.tools.reviewer.localserver.service.Protos.Thread;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.Date;
+
+public class GitHubWriter {
+  private final GitHubClient gitHubClient;
+
+  GitHubWriter(GitHubClient gitHubClient) {
+    this.gitHubClient = gitHubClient;
+  }
+
+  public void writeDiff(Diff diff, String repo) throws IOException {
+    int prNumber = getPRNumber(diff, repo);
+
+    for (Thread diffThread : diff.getDiffThreadList()) {
+      for (Comment comment : diffThread.getCommentList()) {
+        createPullRequestComment(
+            "Created by: **"
+                + comment.getCreatedBy()
+                + "**\nTime: **"
+                + new Date(comment.getTimestamp())
+                + "**\n"
+                + comment.getContent(),
+            prNumber);
+      }
+    }
+
+    for (Thread codeThread : diff.getCodeThreadList()) {
+      for (Comment comment : codeThread.getCommentList()) {
+        String path = codeThread.getFile().getFilename();
+        JSONArray files =
+            new JSONArray(
+                gitHubClient.getResponse(
+                    String.format(
+                        "https://api.github.com/repos/%s/pulls/%d/files", repo, prNumber)));
+        int position = getCommentPosition(codeThread.getLineNumber(), getDiffHunk(files, path));
+        createReviewComment(
+            "Created by: **"
+                + comment.getCreatedBy()
+                + "**\nTime: **"
+                + new Date(comment.getTimestamp())
+                + "**\n"
+                + comment.getContent(),
+            codeThread.getCommitId(),
+            path,
+            position,
+            prNumber);
+      }
+    }
+  }
+
+  private int getPRNumber(Diff diff, String repo) {
+    try {
+      JSONArray repos =
+          new JSONArray(
+              gitHubClient.getResponse(
+                  String.format("https://api.github.com/repos/%s/pulls", repo)));
+      for (Object item : repos) {
+        JSONObject pr = (JSONObject) item;
+        if (pr.getString("title").equals("D" + diff.getId())) {
+          return pr.getInt("number");
+        }
+      }
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    // Creating new PR if it isn't exist
+    return createPullRequest(
+        "D" + diff.getId(), "D" + diff.getId(), "master", diff.getDescription());
+  }
+
+  private int createPullRequest(String title, String head, String base, String body) {
+    JSONObject json = new JSONObject();
+    json.put("title", title);
+    json.put("body", body);
+    json.put("head", head);
+    json.put("base", base);
+
+    String request = "https://api.github.com/repos/val-fed/test-repo/pulls";
```

Comment:
Let's not hard-code `val-fed/test-repo`, and also let's make this static final like in GitHubReader.